src/providers/SignatureProvider.tsx

```diff
@@ -10,7 +11,47 @@ interface InputElementOptions {
 
 const createSignatureProviderState = () => {
     const [imgUrl, setImgUrl] = useState<string>();
+
     const [name, setName] = useState<InputElementOptions>({
+        type: "name",
+        textValue: "",
+        isBold: false,
+        isItalics: false,
+        style: { fontWeight: "normal", fontStyle: "normal" },
+    });
+
+    const [email, setEmail] = useState<InputElementOptions>({
+        type: "email",
+        textValue: "",
+        isBold: false,
+        isItalics: false,
+        style: { fontWeight: "normal", fontStyle: "normal" },
+    });
+
+    const [company, setCompany] = useState<InputElementOptions>({
+        type: "company",
+        textValue: "",
+        isBold: false,
+        isItalics: false,
+        style: { fontWeight: "normal", fontStyle: "normal" },
+    });
+
+    const [title, setTitle] = useState<InputElementOptions>({
+        type: "title",
+        textValue: "",
+        isBold: false,
+        isItalics: false,
+        style: { fontWeight: "normal", fontStyle: "normal" },
+    });
+    const [linkedin, setLinkedin] = useState<InputElementOptions>({
+        type: "linkedin",
+        textValue: "",
+        isBold: false,
+        isItalics: false,
+        style: { fontWeight: "normal", fontStyle: "normal" },
+    });
+    const [phone, setPhone] = useState<InputElementOptions>({
+        type: "phone",
```

Comment:
lots of very similar states being used, could be refactored into one single state? 🤔 💭  @evargast? though I get that such refactoring could mess with all the upper component logic.  But just the one observation on the code pattern.